docs/source/miscellaneous/asynchronous-initialization.rst

```diff
@@ -0,0 +1,32 @@
+===========================
+Asynchronous Initialization
+===========================
```

Comment:
Thanks for writing this! I don't think we documented this properly before (certainly not in such detail)